plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/GenericRewrite.java

```diff
@@ -76,6 +87,13 @@ else if (value instanceof ConnectorExpression) {
                     if (rewritten.isEmpty()) {
                         return Optional.empty();
                     }
+                    if (checkCase && value instanceof Variable variable) {
+                        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(variable.getName());
```

Comment:
Should we verify they should be applied for varchar columns ?

Reply:
To be honest don't know from one point of view introducing this `checkCase` - means, that we just checking case, and we don't care about type.
From other side - check case make sense only for Varchar columns.

I think it depends on decision on https://github.com/trinodb/trino/pull/18140#discussion_r1276077333